lib/utils.js

```diff
@@ -414,7 +416,12 @@ const executeOperation = (topology, operation, args, options) => {
       result => callback(null, result),
       err => callback(err, null)
     );
-    args.push(handler);
+
+    args.push(
+      asyncResource
```

Comment:
oops, this turned out to be incorrect. The Promise case is now wrapped as well, and the bubbleprof visualizations confirm it's working

Reply:
Since we are doing 2 of these:

```js
asyncResource ? asyncResource.runInAsyncScope.bind(asyncResource, handler, undefined) : handler;
```

Can we instead incorporate that into `makeExecuteCallback`?